android/feature/auth/src/main/kotlin/dev/cmux/android/feature/auth/SignInViewModel.kt

```
@@ -0,0 +1,171 @@
+            connection.doOutput = true
+            connection.outputStream.use { it.write(body.toByteArray()) }
+
+            val responseCode = connection.responseCode
+            val responseBody = (if (responseCode in 200..299) connection.inputStream
+            else connection.errorStream)
+                .bufferedReader()
+                .readText()
+            connection.disconnect()
+
+            if (responseCode !in 200..299) {
+                throw Exception("Token exchange failed ($responseCode): $responseBody")
```

detekt (1.23.8)

[warning] Exception is a too generic Exception. Prefer throwing specific exceptions that indicate a specific error case.

(detekt.exceptions.TooGenericExceptionThrown)